packages/provider/src/analytics.ts

```diff
@@ -0,0 +1,50 @@
+import { Databeat, Event as DatabeatEvent, Auth, isBrowser } from '@databeat/tracker'
+
+export enum EventType {
+  // Core types part of Databeat
+  INIT,
+  VIEW,
+
+  // Provider specific
+  SIGN_MESSAGE_REQUEST,
+  SIGN_TYPED_DATA_REQUEST,
+  SEND_TRANSACTION_REQUEST
+}
+
+export type EventTypes = keyof typeof EventType
+export type Event = DatabeatEvent<EventTypes>
+
+// Analytics sub-class to add some custom helper methods
+export class Analytics extends Databeat<EventTypes> {}
+
+// Setup analytics tracker
+export const setupAnalytics = (projectAccessKey: string, server?: string) => {
+  if (!server) {
+    server = 'https://nodes.sequence.app'
+  }
+
+  // disable tracking if projectAccessKey is not set
+  const noop = !(projectAccessKey && projectAccessKey.length > 0)
```

Comment:
Just `const noop = !projectAccessKey` is fine because `''` is falsy and non-empty strings are truthy.